planner/core/task.go

```diff
@@ -914,7 +919,17 @@ func buildIndexLookUpTask(ctx sessionctx.Context, t *copTask) *rootTask {
 	// (indexRows / batchSize) * batchSize * CPUFactor
 	// Since we don't know the number of copTasks built, ignore these network cost now.
 	indexRows := t.indexPlan.statsInfo().RowCount
-	newTask.cst += indexRows * sessVars.CPUFactor
+	idxCst := indexRows * sessVars.CPUFactor
+	// if the expectCnt is below the paging threshold, using paging API, recalculate idxCst.
+	// paging API reduces the count of index and table rows, however introduces more seek cost.
+	if ctx.GetSessionVars().EnablePaging && t.expectCnt > 0 && t.expectCnt <= paging.Threshold {
+		p.Paging = true
+		// we want the diff between idxCst and pagingCst here,
+		// however, the idxCst does not contain seekFactor, so a seekFactor needs to be removed
+		pagingCstDiff := calcPagingCost(ctx, t) - sessVars.GetSeekFactor(nil)
+		idxCst = math.Min(idxCst, pagingCstDiff)
```

Comment:
why is the idxCst = math.Min(idxCst, pagingCstDiff)? 
If choose paging reading, the cost of index should be calculated by paging which value is `calcPagingCost(ctx, t) - sessVars.GetSeekFactor(nil)`.
I think the variable `pagingCstDiff` should be renamed to `pagingCost`, because the `pagingCstDiff` means the difference of paging cost, but here we want a paging cost

Reply:
There are some cases that don't work well because of the statistic error, e.g. if we use paging, however, `pagingCst > idxCst`, we enlarge the cost of this plan. As a result, another plan may be chosen by the CBO at last.

Reply:
IMO, `pagingCost := calcPagingCost(ctx, t)`, the variable `pagingCstDiff` stands for `pagingCost - seekCnt`, that means the diff between `pagingCst` and one seek cost in the below data source plan.